premake5.lua

```diff
@@ -40,6 +40,102 @@
+				local exists = os.isfile(docfilepath)
+				if not exists then
+					print("Genearting documenetation file for: ", k)
+					count = count + 1
+
+					-- Create doc file
+					local docfile = io.open(docfilepath, "w")
+					docfile:write(k)
+					docfile:write("\n\n")
+					docfile:write("```lua\n")
+					docfile:write(k .. " (value)\n")
+					docfile:write("```\n\n")
+					docfile:write("### Parameters ###\n\n")
+					if type(v.allowed) == "table" then
+						docfile:write("`value` is one of:\n\n")
+						for _,value in ipairs(v.allowed) do
+							docfile:write("* `" .. value .. "`: needs documentation\n")
+						end
+						docfile:write("\n")
+					else
+						docfile:write("`value` - needs documentation.\n\n")
+					end
+					docfile:write("## Applies To ###\n\n")
+					if #v.scopes == 1 then
+						docfile:write("The `" .. v.scope .. "` scope.\n\n")
+					else
+						docfile:write("The ")
+						for i,scope in ipairs(v.scopes) do
+							if i > 1 then
+								docfile:write(", ")
+							end
+							docfile:write("`" .. scope .. "`")
+						end
+						docfile:write(" scopes.\n\n")
+					end
+					docfile:write("### Availability ###\n\n")
+					-- Get git commits that contains API registration
+					local gitlog = "git log origin/master -S\"name = \\\"" .. k .. "\\\"\" --source --reverse --pretty=format:\"%h\""
+					-- Override path.normalize so it stops mangling our commands!
+					local pathnormalize = path.normalize
+					path.normalize = function (v) return v end
+					local stage = ""
+					local requiredversion = 0
+					local commits = os.outputof(gitlog)
+					local tags = ""
+					if commits ~= nil and commits ~= "" then
+						local commit = string.explode(commits, "\n")[1]
+						-- Get git tags that contain first commits
+						tags = os.outputof("git tag --contains " .. commit)
+					end
+
+					-- Get latest tag if commit hasn't been tagged yet
+					if tags == nil or tags == "" then
+						tags = os.outputof("git tag --sort=-creatordate")
+					end
+					local tag = string.explode(tags, "\n")[1]
+					-- tag should be in the format of `v5.0.0-alphaXX` where XX is the alpha version
+					local subversion = string.explode(tag, "-")[2]
+					-- subversion may be nil if there's no tags checked out
+					if subversion ~= nil then
+						if subversion:startswith("alpha") then
+							stage = "alpha"
+						elseif subversion:startswith("beta") then
+							stage = "beta"
+						end
+						subversion = subversion:sub(stage:len() + 1)
+						requiredversion = tonumber(subversion) + 1
+						-- There was no alpha17, instead specify beta1
+						if stage == "alpha" and requiredversion == 17 then
+							stage = "beta"
+							requiredversion = 1
+						end
+						docfile:write("Premake 5.0.0 " .. stage .. " " .. requiredversion .. " or later.\n\n")
+					else
+						docfile:write("Premake 5.0.0 unknown or later.\n\n")
+					end
+					path.normalize = pathnormalize
+					docfile:write("### Examples ###\n\n")
```

Comment:
Not sure about the (dummy) example section (not all existing doc contains example).
Especially for the ones when `allowed` is a `table`.

Commented dummy section "See also" might be useful for consistency.

Wonder if `register` should not have a "description" field (as the new option has) to.

Reply:
> Not sure about the (dummy) example section (not all existing doc contains example). Especially for the ones when `allowed` is a `table`.
> 
> Commented dummy section "See also" might be useful for consistency.

I'm not sure which API I was using as the template, but I believe it was [this one](https://github.com/premake/premake-core/blob/master/website/docs/debugger.md). Which based on the [wiki commit](https://github.com/premake/premake-core/wiki/_compare/d407be92ee2f8cd0030beb177bbece5d5199f124) that added it, looks like it was just a bunch of auto-generated files. This is the template that we've done before, so I figured it was good to continue using it.

As for removing examples and adding "see also", I just don't quite understand why we would do that? In my opinion, not every API requires a "see also", but most APIs should probably have an example of their usage. Even if that example is just showing an allowed value being used. I don't think this should be part of the auto-gen docs though, I believe the docs writer needs to make a call on which value should be used as an example, with an appropriate description of how it's being used. For example, [`disablewarnings`](https://premake.github.io/docs/disablewarnings#examples) this shows how you can actually use it, not just calling the API blindly, as well as providing a description of what the example does. [`compileas`](https://premake.github.io/docs/compileas#examples) is another example of using the API, this time with a set of allowed values, however it doesn't contain a description of what the example is doing. I think a placeholder example is enough to get doc writers going, but perhaps the placeholder needs more guiding text?

> Wonder if `register` should not have a "description" field (as the new option has) to.

I'm not quite sure this is relevant to this PR? This should probably be it's own issue/discussion instead.

Reply:
I meant generated files would requires someone to fix content anyway, so I proposed
```
<!-- 
Example:
...
See Also:
-->
```
So not visible as default, but allow the ones who fix content to respect the template.
but nvm.